package/base-files/files/lib/preinit/10_indicate_preinit

```diff
@@ -63,6 +63,24 @@ preinit_config_switch() {
 	json_select ..
 }
 
+preinit_config_port() {
+	local original
+
+	local port=$1
+	local path=$2
+
+	[ -d "/sys/devices/$path/net" ] || return
+	original="$(ls /sys/devices/$path/net | head -1)"
+
+	[ "$port" = "$original" ] && return
+
+	if [ -h "/sys/class/net/$port" ]; then
+		ip link set $port down
+		ip link set $port name $(echo "$path" | md5sum | head -c 16)
```

Comment:
This should probably be `head -c 8`: if e.g. a USB network device is added but not remapped, and then subsequently *used*, it could end up too long.

Also, the name of the port is not based on a hash of the path, but on the sysfs path of its replacement. @jow- made the comment that this reconfiguration should be done in two passes: here's one more reason why.

Reply:
do you mind implementing the two step approach?

Reply:
Regarding the attached checksum, I think `-c 4` should be just fine, how many network interfaces do you expect to have>

Reply:
I think 4 is too few -- you run the risk of running into the birthday problem eventually with hashes: With 12 ports, if all need to be renamed, there's just over a 0.1% chance of a collision among 16 bits; with 32 bits it's closer to 1 in a million.

Still, this is silly: if we're doing this in two passes we can just increment the paths starting from the highest eth at that point; if we detect eth0-11, we can start by moving to eth12, then eth13, etc.

Reply:
@aparcar @jow-  I've implemented the two-step approach as of 78bc40cf3d114aba019871c9f18d1174fe38b61a. Note that I no longer bother with randomization, instead finding the highest ethernet netdev number and adding one more for temporary interfaces.